boards/arm/ngultra_devkit/doc/index.rst

```diff
@@ -0,0 +1,80 @@
+.. _ngultra_devkit:
+
+NanoXplore NG-Ultra Devkit
+############################
+
+Overview
+********
+
+The NG-Ultra devkit is a demonstrator board showing NG-Ultra capabilities.
+The NG-Ultra is a radiation hardened by design SoC for aerospace applications.
+
+.. figure:: ngultra_devkit.jpg
+   :align: center
+   :alt: NG-Ultra Devkit
+
+Hardware
+********
+
+- 4 ARM Cortex-R52
+- hardened FPGA fabric (540k LUT)
+- 6 status LEDs
+- 256 MiB of flash memory
+- 256 MiB of DDR2 memory with redundancy
+- SpaceWire bus
+
+Supported Features
+==================
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| GIC-v3    | on-chip    | interrupt controller |
++-----------+------------+----------------------+
+| UART      | on-chip    | serial               |
++-----------+------------+----------------------+
+
+Other hardware features are not currently supported by Zephyr.
+
+Connections and IOs
+===================
+
+J28 Mictor-38 connector
+
++-------+--------------+
+| PIN # | Signal Name  |
++=======+==============+
+| 3     | TRST         |
++-------+--------------+
+| 5     | TDI          |
++-------+--------------+
+| 7     | TMS          |
++-------+--------------+
+| 9     | TCK          |
++-------+--------------+
+| 13    | TDO          |
++-------+--------------+
+| 15    | SRST         |
++-------+--------------+
+| 20    | GND          |
++-------+--------------+
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+The board is delivered with an SDK which provides scripts to flash the
+device.
```

Comment:
? what about `west flash`? How is generated `zephyr.bin/hex` expected to be used?

Reply:
Today our SDK could flash zephyr or our baremetal BSP with the same scripts. If `west flash` is mandatory, I can give a try but it won't be easy because of the OTP.
A bl0 could load the bin from flash otherwise zephyr.bin/hex could be loaded from OpenOCD to eram.

Reply:
Ping

Reply:
When you are talking about programming support, are you talking about a tool to flash zephyr to the board?

Reply:
If, for example, openocd is supported, then just add support for it, users will automatically be able to use standard flashing procedure, ie `west flash` instead of relying on custom scripts. If you have another tool with extra features (as many vendors do, e.g. ST supports openocd and stcubeprogrammer) you can implement a runner for it, so that west flash can be used as well. The board documentation doesn't even provide a link to such SDK scripts...

Reply:
I understand your concerns about standard flashing, I will work on a board.cmake to provide at least a runner. I will add a link to [NanoXplore gitlabe](https://gitlabext.nanoxplore.com) & confluence with [flashing instructions](https://nanoxplore-wiki.atlassian.net/wiki/spaces/NAN/pages/135626753/NanoXplore+NG-ULTRA+Appnote+BL1+flashmem)

Reply:
Still pending. I see the documentation mentions openocd, maybe you can reuse the openocd target?